src/rgw/rgw_rados.cc

```diff
@@ -11952,9 +11965,38 @@ int RGWRados::get_bucket_info(RGWObjectCtx& obj_ctx,
     ldout(cct, 20) << "couldn't put binfo cache entry, might have raced with data changes" << dendl;
   }
 
+  if (refresh_version &&
+      refresh_version->compare(&info.objv_tracker.read_version)) {
+    lderr(cct) << "WARNING: The OSD has the same version I have. Something has gone "
+               << "squirrelly, since this function should only be called "
+               << "in the event that an operation is rejected by "
+               << "version_check on the OSD." << dendl;
+    return -EALREADY;
```

Comment:
do we want to even return an error here? is it an error? Maybe there was some kind of a forced  change that happened outside that ended up resetting the version. Whatever we read now should be fine, only need to warn.

Reply:
Can we have those kinds of forced changes? That is, is there any way where we can have the OSD tell that we have an invalid version, then we read direct from the OSD and it gives us the same version we had before? I would have figured that's an impossible situation, but if it can come up I'll remove the error return.

Reply:
Or at least I'd been working under the assumption that the versions must /always/ be monotonically increasing. I hadn't thought something external would be allowed to ratchet the version number down like that.

Reply:
The osd is not managing these versions, only verifies that they match (as requested by rgw). While not common, a version could be overriden (e.g., user injects object metadata).